llvm/lib/Target/AMDGPU/AMDGPUISelDAGToDAG.cpp

```diff
@@ -1926,7 +1926,7 @@ bool AMDGPUDAGToDAGISel::checkFlatScratchSVSSwizzleBug(
   KnownBits VKnown = CurDAG->computeKnownBits(VAddr);
   KnownBits SKnown =
       KnownBits::add(CurDAG->computeKnownBits(SAddr),
-                     KnownBits::makeConstant(APInt(32, ImmOffset)));
+                     KnownBits::makeConstant(APSInt(32, ImmOffset)));
```

Comment:
I'd suggest leaving it as `APInt` but passing `/*isSigned=*/ true` as the third argument.